protocol/message_persistence.go

```diff
@@ -2207,3 +2181,161 @@ func (db sqlitePersistence) SetContactRequestState(id string, state common.Conta
 	_, err := db.db.Exec(`UPDATE user_messages SET contact_request_state = ? WHERE id = ?`, state, id)
 	return err
 }
+
+func getUpdatedChatMessagePayload(originalMessage *protobuf.DiscordMessage, attachmentMessage *protobuf.DiscordMessage) *protobuf.ChatMessage_DiscordMessage {
+	originalMessage.Attachments = append(originalMessage.Attachments, attachmentMessage.Attachments...)
+	return &protobuf.ChatMessage_DiscordMessage{
+		DiscordMessage: originalMessage,
+	}
+}
+
+func getMessageFromScanRows(db sqlitePersistence, rows *sql.Rows) (*common.Message, error) {
+	var msg *common.Message
+
+	for rows.Next() {
+		// There's a possibility of multiple rows per message if the
+		// message has a discordMessage and the discordMessage has multiple
+		// attachments
+		//
+		// Hence, we make sure we're aggregating all attachments on a single
+		// common.Message
+		var message common.Message
+		err := db.tableUserMessagesScanAllFields(rows, &message)
+		if err != nil {
+			return nil, err
+		}
+
+		if msg == nil {
+			msg = &message
+		} else if discordMessage := msg.GetDiscordMessage(); discordMessage != nil {
+			msg.Payload = getUpdatedChatMessagePayload(discordMessage, message.GetDiscordMessage())
+		}
+	}
+	return msg, nil
+}
+
+type Messages []*common.Message
+type PinnedMessages []*common.PinnedMessage
```

Comment:
I would also move these declarations into the `common` package so that they can benefit from the some importability that the base structs have.